hooks/rnc-format-validate.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+import subprocess
+from pathlib import Path
+
+
+def main():
+
+    proc = subprocess.Popen(
+        ["git", "diff", "--name-only", "--cached"], stdout=subprocess.PIPE
+    )
+    changed_files = proc.stdout.readlines()
+    changed_files = [f.decode("utf-8") for f in changed_files]
+    changed_files = [f.strip() for f in changed_files]
+
+    # I don't understand why, but this currently isn't working
+    for rncfile in changed_files:
+        if rncfile.endswith(".rnc"):
+            rngfile = Path(rncfile).stem
+            subprocess.call(["trang", rncfile])
+            subprocess.call(["trang", rngfile, rncfile])
+            subprocess.call(["git", "add", rncfile])
```

Comment:
`subprocess.check_call` to ensure there's a zero exit status.... not sure if that's what you want for the hook

Reply:
The idea is that the script fixes the formatting and finishes the commit without failing.

You're suggesting here changing all `call` -> `check_call`?

Reply:
If you want the script to fail if the subprocess calls fail, then I'd use `.check_call`. Otherwise, I'd use `.run`.

Reply:
The issue I was having with this was the checks were passing, and the commit made, but the trang formatting wasn't applied.

Reply:
probably git adding `Path(rncfile).stem` doesn't work. Should add full filename including suffixes

Reply:
I'm just trying to the basename there, but I guess we don't need human readable filenames for this!

Reply:
Fixed.